staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go

```diff
@@ -379,3 +379,38 @@ func TestPreserveInt(t *testing.T) {
 		t.Errorf("Expected %v, got %v, %v", `9223372036854775807, 1000000`, num1, num2)
 	}
 }
+
+func TestNameCollision(t *testing.T) {
+	stopCh, apiExtensionClient, clientPool, err := testserver.StartDefaultServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer close(stopCh)
+
+	noxuDefinition := testserver.NewNoxuCustomResourceDefinition(apiextensionsv1alpha1.NamespaceScoped)
+	_, err = testserver.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, clientPool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	noxu2Definition := testserver.NewNoxu2CustomResourceDefinition(apiextensionsv1alpha1.NamespaceScoped)
+	_, err = testserver.CreateNewCustomResourceDefinition(noxu2Definition, apiExtensionClient, clientPool)
+	if err == nil {
```

Comment:
This takes 30sec. I would prefer to ignore discovery here and and just do a `wait.Poll` checking the status for the `NameConflict`.